tests/components/ruuvi_gateway/test_config_flow.py

```diff
@@ -0,0 +1,133 @@
+"""Test the Ruuvi Gateway config flow."""
+from unittest.mock import patch
+
+from aioruuvigateway.excs import CannotConnect, InvalidAuth
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components import dhcp
+from homeassistant.components.ruuvi_gateway.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult, FlowResultType
+
+from .consts import (
+    BASE_DATA,
+    EXPECTED_TITLE,
+    GATEWAY_MAC_LOWER,
+    GET_GATEWAY_HISTORY_DATA,
+)
+from .utils import patch_gateway_ok, patch_setup_entry_ok
+
+DHCP_IP = "1.2.3.4"
+DHCP_DATA = {**BASE_DATA, "host": DHCP_IP}
+
+
+@pytest.mark.parametrize(
+    "init_data, init_context, entry",
+    [
+        (
+            None,
+            {"source": config_entries.SOURCE_USER},
+            BASE_DATA,
+        ),
+        (
+            dhcp.DhcpServiceInfo(
+                hostname="RuuviGateway1234",
+                ip=DHCP_IP,
+                macaddress="12:34:56:78:90:ab",
+            ),
+            {"source": config_entries.SOURCE_DHCP},
+            DHCP_DATA,
+        ),
+    ],
+    ids=["user", "dhcp"],
+)
+async def test_ok_setup(hass: HomeAssistant, init_data, init_context, entry) -> None:
+    """Test we get the form."""
+    init_result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        data=init_data,
+        context=init_context,
+    )
+    assert init_result["type"] == FlowResultType.FORM
+    assert init_result["step_id"] == config_entries.SOURCE_USER
+    assert init_result["errors"] is None
+
+    # Check that setup is okay
+    await assert_finalize_setup(hass, entry, init_result)
```

Comment:
From a maint perspective want the assertions in the test itself so we don't have to dig for the failure path when something fails.

You could also do `pytest.register_assert_rewrite` to accomplish this but in practice its probably easier to copy them.

Reply:
In my opinion, this is clear enough (and you could clearly see the failure path in the pytest tracebacks anyway?).

If you insist, I can inline the `assert_finalize_setup` function, but I have something against copy-pasting the same thing and inevitably ending up changing it in four places later...

Reply:
Please adjust. thanks!